backend/RoomPlanner/room-planner/src/main/java/edu/roomplanner/rest/AuthController.java

```diff
@@ -0,0 +1,78 @@
+package edu.roomplanner.rest;
+
+import edu.roomplanner.dto.CustomUserDetails;
+import edu.roomplanner.entity.PersonEntity;
+import edu.roomplanner.repository.PersonRepository;
+import edu.roomplanner.dto.AuthenticationRequest;
+import edu.roomplanner.service.impl.CustomUserDetailsService;
+import edu.roomplanner.security.JwtTokenProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.*;
+
+import static org.springframework.http.ResponseEntity.ok;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/auth")
+public class AuthController {
+
+    private AuthenticationManager authenticationManager;
+    private JwtTokenProvider jwtTokenProvider;
+    private PersonRepository persons;
+    private CustomUserDetailsService customUserDetailsService;
+
+    @Autowired
+    public AuthController(AuthenticationManager authenticationManager,
+                          JwtTokenProvider jwtTokenProvider,
+                          PersonRepository persons,
+                          CustomUserDetailsService customUserDetailsService) {
+        this.authenticationManager = authenticationManager;
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.persons = persons;
+        this.customUserDetailsService = customUserDetailsService;
+    }
+
+    @GetMapping("/test")
+    public String test() {
+        return "ok";
+    }
+
+    @PostMapping("/signin")
+    public ResponseEntity signin(@RequestBody AuthenticationRequest data) {
+        try {
+            String email = data.getEmail();
```

Comment:
This logic should be extracted into a service. It should not stay in the controller.